[Feat] 상품이미지 업로드 로직 아마존 S3 연동

juhee200 opened the pull request:
## 기능 설명
상품이미지 업로드 로직 아마존 S3 연동

## 작업 내용
- ItemService 이미지 클라우드 업로드 로직 추가
- 클라우드 이미지 주소 db 저장

## 수정 사항

## 추가 작업 예정

juhee200 committed:
FEAT: Item Image S3 Cloud Service

- feat: upload item images in S3 cloud

Related to: #23

juhee200 committed:
FEAT: Item Image S3 Cloud Service

 feat: upload item images in S3 cloud

Related to: #23

POKUDING commented on src/main/java/com/poku/graypants/domain/item/application/ItemService.java:
```diff
@@ -1,36 +1,62 @@
+        List<MultipartFile> multipartItemPhotos = itemCreateRequestDto.getItemPhotos();
+        MultipartFile multipartItemDesc = itemCreateRequestDto.getItemDescImg();
+
+        Item savedItem = itemRepository.save(itemCreateRequestDto.toEntity());
```
save 이후 savedItem 을 수정하는 것 보다 item 을 toEntity로 만들고 사진을 업데이트한 이후 save를 하는게 더 좋아보입니다.

jgoneit replied:
동의합니다 :)

juhee200 replied:
이 부분에 대해서 itemphoto 테이블에 item이 필요해서 저장 후 업데이트를 하는 것을 선택했는데 itemPhoto 테이블을 삭제하고 String으로 Url 저장을 해서 json으로 파싱하는 방법으로 하겠습니다.

POKUDING commented on src/main/java/com/poku/graypants/domain/item/application/ItemService.java:
```diff
@@ -54,4 +80,68 @@ private Item getItemByID(Long id) {
+        try {
+            multipartFile.transferTo(file);
+        } catch (IOException e) {
+            throw new RuntimeException(ExceptionStatus.FILE_NOT_FOUND.getMessage());
```
RuntimeException은 핸들러가 없으니 사용을 지양해주시기 바랍니다.

juhee200 replied:
넵 수정하겠습니다.

juhee200 committed:
ADD: merge to dev

Jinseonghyun reviewed (approved):
수고하셨습니다~

POKUDING reviewed (commented):
고생하셨습니다!

juhee200 committed:
FEAT: itemPhoto Table delete

-Refactoring: delete itemPhoto table, instead another table use  List<String>

Related to: #23

juhee200 committed:
Merge branch 'dev' into feat-#23-hee
